scadnano/scadnano.py

```diff
@@ -6946,6 +6946,80 @@ def _write_plates_default(self, directory: str, filename: Optional[str], strands
 
         workbook.save(filename_plate)
 
+    def to_oxview_format(self, warn_duplicate_strand_names: bool = True, use_strand_colors: bool = True) -> dict:
+        """
+        Exports to oxView format.
+
+        :param warn_duplicate_strand_names:
+            if True, prints a warning to the screen indicating when strands are found to
+            have duplicate names. (default: True)
+        :param use_strand_color:
+            if True (default), sets the color of each nucleotide in a strand in oxView to the color
+            of the strand.        
+        """
+        import datetime
+        self._check_legal_design(warn_duplicate_strand_names)
+        system = _convert_design_to_oxdna_system(self)
+
+        oxview_strands: List[Dict[str, Any]] = []
+        nuc_count = 0
+        strand_count = 0
+        for sc_strand, oxdna_strand in zip(self.strands, system.strands):
+            strand_count += 1
+            oxvnucs: List[Dict[str, Any]] = []
+            oxvstrand = {'id': strand_count, 
+                         'class': 'NucleicAcidStrand', 
+                         'end5': nuc_count, 
+                         'end3': nuc_count+len(oxdna_strand.nucleotides), 
+                         'monomers': oxvnucs}
+            if use_strand_colors and (sc_strand.color is not None):
+                scolor = sc_strand.color.to_cadnano_v2_int_hex()
+            else:
+                scolor = None
+            
+            for index_in_strand, nuc in enumerate(oxdna_strand.nucleotides):
+                oxvnuc = {'id': nuc_count, 
```

Comment:
There's also a field `bp` for base pair. It would be good to look this up in the design and assign it here. Right now I can't remember if `Design` has a simple method to do this, but the code in [`Design.assign_dna_complement_from`](https://github.com/UC-Davis-molecular-computing/scadnano-python-package/blob/724a4a8f75b681a1438451ae430a083b1e7a27a5/scadnano/scadnano.py#L3492) should have similar logic.

Reply:
Ah, this would be quite useful, yes.  It's a bit tricky, as it needs to use the oxDNA strand/nucleotide numbering.  It would also probably make sense to have this implemented for oxDNA export as a forces file, which seems to be the main purpose of the 'bp' field in oxView.  I'll see what I can do this evening.

Reply:
I've added code specific to the oxview format to export base pairs.  I might move this to something separate later, or just replicate it, for oxDNA, but for now it works.

Reply:
Awesome, thank you. But I thought oxDNA doesn't have any stored notion of base pairs?

Reply:
It doesn't, directly.  However, it seems like a common practice for relaxation of structures far from a stable configuration (eg, something exported from scadnano) is to include an external forces file that adds mutual_trap forces between intended base pairs, holding the structure together as it relaxes.  This seems to be the only purpose at the moment for the bp values in oxview, for example.  Doing this manually would be infeasible, and exporting desired base pairs (as we do here for oxview) seems much safer than the automatic base pair detection in oxdna_analysis_tools.